misc/toml-parser/src/main/java/io/ballerina/toml/validator/SchemaValidator.java

```diff
@@ -130,13 +148,31 @@ public void visit(TomlValueNode tomlValue) {
         visitNode(tomlValue);
     }
 
+    private String getTypeErrorMessage(Type found) {
+        Map<String, String> message = this.schema.message();
+        String typeCustomMessage = message.get(SchemaDeserializer.TYPE);
+        if (typeCustomMessage == null) {
+            return String.format("key '%s' expects %s . found %s", this.key, schema.type(), found);
```

Comment:
How about something like,
```
incompatible type for key 'foo': expected 'T1', found 'T2'
```

Reply:
@pubudu538 ATM T1 values are all capital. eg: `ARRAY`. Shall we make it like `array`?